articles/azure/azs-how-to-retrieve-billing-api.md

```diff
@@ -0,0 +1,219 @@
+
+# Variables
+$domain = "common"
+$resource = 'https://api.partnercenter.microsoft.com'
+$clientid = "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
+$username = "XXXXXXXXX@XXXXXXXXX.com"
+$password = 'XXXXXXXXXXXX'
+
+# Obtain AAD Token
+$url  = "https://login.windows.net/{0}/oauth2/token" -f $domain
+$body = "grant_type=password&"
+$body = $body + "client_id=$clientid&"
+$body = $body + "resource=$resource&"
+$body = $body + "username=$username&"
+$body = $body + "password=$password&"
+$body = $body + "scope=openid"
+ 
+$response = Invoke-RestMethod -Uri $url -Body $body -method "POST" -Headers $headers
+$response.access_token
+$aadtoken = $response.access_token
+ 
+# Obtain CSP token based on AAD token
+$url  = "https://api.partnercenter.microsoft.com/generatetoken"
+$body = "grant_type=jwt_token"
+$headers=@{Authorization="Bearer $aadtoken"} 
+
+$responseCSPToken = Invoke-RestMethod -Uri $url -Headers $headers -Body $body -method "POST" 
+$responseCSPToken.access_token 
+
+$headerstest= @{Authorization="Bearer  $($responseCSPToken.access_token)"}
+
+# Query CSP Portal
+$CSPQuery = Invoke-WebRequest -uri "https://api.partnercenter.microsoft.com/v1/invoices/<"Insert Invoice Name">/documents/statement" -UseBasicParsing -Headers $headerstest -OutFile c:\temp\test1.pdf
+$CSPQuery
+
+#########################################
+```
+
+### Bash
+
+```bash
+#########################################
+
+ # Variables
+ domain='common'
+ resource='https://api.partnercenter.microsoft.com'
+ clientid = 'XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX'
+ username = 'XXXXXXXXX@XXXXXXXXX.com'
+ password = 'XXXXXXXXXXXX'
+  
+ # Obtain AAD Token
+ ADToken=$(curl -X "POST" "https://login.windows.net/common/oauth2/token" \
+ -H "Content-Type: application/x-www-form-urlencoded" \
+ --data-urlencode "client_id=$clientid" \
+ --data-urlencode "grant_type=password" \
+ --data-urlencode "username=$username" \
+ --data-urlencode "password=$password" \
+ --data-urlencode "resource=$resource"  \
+  |grep -o -P '(?<=access_token":").*(?=","refresh_token)')
+ 
+ # Obtain CSP token based on AAD token
+ CSPToken=$(curl -X "POST" "https://api.partnercenter.microsoft.com/generatetoken" \
+ -H "Content-Type: application/x-www-form-urlencoded" \
+ -H "Authorization: Bearer $ADToken" \
+ --data-urlencode "grant_type=jwt_token" \
+ | grep -o -P '(?<=access_token":").*(?=","token_type)')
+
+ # Query CSP Portal  
+ curl -X "GET" "https://api.partnercenter.microsoft.com/v1/invoices/<"Insert Invoice Name">/lineitems/Azure/BillingLineItems" \
+ -H "Content-Type: application/json" \
+ -H "Accept: application/json" \
+ -H "Authorization: Bearer $CSPToken" \
+ > billing.csv
+
+ #########################################
+ ```
+## Feedback
+
+If you find an issue with this article, click Improve this Doc to suggest a change. If you have an idea for how we could improve any of our services, visit UKCloud Ideas. Alternatively, you can contact us at products@ukcloud.com.
```

Comment:
Make Improve this Doc bold
Add a hyperlink to UKCloud Ideas that points to https://ideas.ukcloud.com
Put angled brackets (<>) around products@ukcloud.com

Reply:
fixed